File: src/SkyBlock/enchants/bow/player/Shuffle.php
<?php


namespace SkyBlock\enchants\bow\player;


use pocketmine\event\entity\EntityDamageByEntityEvent;
use pocketmine\player\Player;

class Shuffle extends BaseBowHitPlayerEnchant {

    static int $id = 121;

    /**
     * @param Player                    $attacker
     * @param Player                    $victim
     * @param int                       $enchantmentLevel
     * @param EntityDamageByEntityEvent $event
     */
    public function onActivation(Player $attacker, Player $victim, int $enchantmentLevel, EntityDamageByEntityEvent $event) : void {
        $pos1 = clone $attacker->getLocation();
        $pos2 = clone $victim->getLocation();
        if ($pos1->getWorld()->getDisplayName() === 'PvP' && $pos2->getWorld()->getDisplayName() === 'PvP') {
            if ($pos1->distance($pos2) <= 50) {
                $attacker->teleport($pos2->add(0, 1, 0));
                $victim->teleport($pos1->add(0, 1, 0));
                $this->sendActivation($attacker, "§bShuffle §aActivated!");
                $this->sendActivation($victim, "§cStruck by §bShuffle §cEnchant!");
                return;
            }
        }
        $this->sendActivation($attacker, "§bShuffle §cDeactivated, player too far!");
    }

}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/bow/Shuffle.java
package dev.skyblock.enchants.bow;
import dev.skyblock.enchants.BaseBowEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
public class Shuffle extends BaseBowEnchant {
    public static final int ID = 121;
    public Shuffle() { super(ID, "Shuffle", 5); }
    @Override
    public void onHitPlayer(Player shooter, Player hit, EntityDamageByEntityEvent ev, int level) {

    }
}
